Modified language to better adhere to IBM style guide (#68)
devfile/docs: docs/modules/user-guide/partials/proc_defining-kubernetes-resources.adoc

= Defining Kubernetes resources

[role="_abstract"]

> Use Kubernetes or OpenShift resource lists to describe complex deployments. Use devfiles to reference Kubernetes or the OpenShift resource lists; referencing them makes them part of the workspace.

Reviewer: ```suggestion
Reference Kubernetes or OpenShift resource lists in devfiles to describe complex deployments. This them makes them part of the workspace.
```

.Procedure

[IMPORTANT]
====

> * Because a devfile workspace is internally represented as a single deployment, all resources from the Kubernetes or OpenShift list are merged into that single deployment.

Reviewer: `{prod-short}` != devfile

Author: Didn't I change it back to `{prod-short}` in my recent commit? 

In a later issue, we should change the {prod-short} file to be lowecase, `devfile` not `Devfile`. We don't capitalize devfile unless it starts a sentence. The authoring stacks use `{prod-short}` more than a lot of our other docs, so since we're keeping it here, we'll want to fix this capitalization inconsistency.

Reviewer: > Didn't I change it back to {prod-short} in my recent commit?

That's the point. You did, which is wrong. It resolves to "devfiles workspace', which makes no sense in this context. In Che docs, it used to resolve to "Che workspace", which is a thing. The docs were copied without being properly adapted.

> In a later issue, we should change the {prod-short} file to be lower case, devfile not Devfile. We don't capitalize devfile unless it starts a sentence.

Ack.

> The authoring stacks use {prod-short} more than a lot of our other docs

Clearly :/

> so since we're keeping it here, we'll want to fix this capitalization inconsistency.

No, we're not keeping because it make no sense. It's just a left-over from when it was a Che-specific doc.

> * Be careful when designing such lists because they can produce name conflicts.

Reviewer: The lists produce nothing. The user who designs them does.

```suggestion
* Avoid name conflicts when designing such lists.
```

* Only the following subset of the Kubernetes objects are supported:

** `deployments`
** `pods`
** `services`
** `persistent volume claims`
** `secrets`
** `config maps`

Kubernetes Ingresses are ignored, but OpenShift routes are supported. A workspace created from a devfile using any other object types fails to start.

> * When running devfiles on a Kubernetes cluster, only Kubernetes lists are supported. When running devfiles on an OpenShift cluster, both Kubernetes and OpenShift lists are supported.

Reviewer: `{prod-short}` != devfile

      uri: ../relative/path/postgres.yaml
----

> The preceding component references a file that is relative to the location of the devfile. Therefore, this devfile is only loadable by a devfile factory. You supply the location of your devfile to the devfile factory, and then your devfile figures out the location of the referenced Kubernetes resource list.

Reviewer: * `{prod-short}` != devfile (There are no "devfile factories".)

* Don't use phrasal verbs (_to figure out_).
